Source/Parallelization/WarpXComm_K.H

```diff
@@ -39,30 +39,30 @@ void warpx_interp (int j, int k, int l,
     const int sl = (AMREX_SPACEDIM <= 2) ? 0 : arr_stag[2];
 
     // Number of points used for interpolation from coarse grid to fine grid
-    const int nj = (sj == 0) ? 1 : 2;
-    const int nk = (sk == 0) ? 1 : 2;
-    const int nl = (sl == 0) ? 1 : 2;
+    const int nj = 2;
+    const int nk = 2;
+    const int nl = 2;
 
-    const int jc = amrex::coarsen(j, rj);
-    const int kc = amrex::coarsen(k, rk);
-    const int lc = amrex::coarsen(l, rl);
+    const int jc = (sj == 0) ? amrex::coarsen(j - rj/2, rj) : amrex::coarsen(j, rj);
+    const int kc = (sk == 0) ? amrex::coarsen(k - rk/2, rk) : amrex::coarsen(k, rk);
+    const int lc = (sl == 0) ? amrex::coarsen(l - rl/2, rl) : amrex::coarsen(l, rl);
 
     amrex::Real wj;
     amrex::Real wk;
     amrex::Real wl;
 
-    // Interpolate from coarse grid to fine grid using either 1 point with weight 1, if both grids
-    // are cell-centered, or 2 points with weights depending on the distance, if both grids are nodal
+    // Interpolate from coarse grid to fine grid using 2 points
+    // with weights depending on the distance, for both nodal and cell-centered grids
     amrex::Real res = 0.0_rt;
     for         (int jj = 0; jj < nj; jj++) {
         for     (int kk = 0; kk < nk; kk++) {
             for (int ll = 0; ll < nl; ll++) {
-                wj = (sj == 0) ? 1.0_rt : (rj - amrex::Math::abs(j - (jc + jj) * rj))
-                                          / static_cast<amrex::Real>(rj);
-                wk = (sk == 0) ? 1.0_rt : (rk - amrex::Math::abs(k - (kc + kk) * rk))
-                                          / static_cast<amrex::Real>(rk);
-                wl = (sl == 0) ? 1.0_rt : (rl - amrex::Math::abs(l - (lc + ll) * rl))
-                                          / static_cast<amrex::Real>(rl);
+                wj = (sj == 0) ? (rj - amrex::Math::abs(j + 0.5_rt - (jc + jj + 0.5_rt) * rj)) / static_cast<amrex::Real>(rj)
+                               : (rj - amrex::Math::abs(j          - (jc + jj         ) * rj)) / static_cast<amrex::Real>(rj);
+                wk = (sk == 0) ? (rk - amrex::Math::abs(k + 0.5_rt - (kc + kk + 0.5_rt) * rk)) / static_cast<amrex::Real>(rk)
+                               : (rk - amrex::Math::abs(k          - (kc + kk         ) * rk)) / static_cast<amrex::Real>(rk);
+                wl = (sl == 0) ? (rl - amrex::Math::abs(l + 0.5_rt - (lc + ll + 0.5_rt) * rl)) / static_cast<amrex::Real>(rl)
+                               : (rl - amrex::Math::abs(l          - (lc + ll         ) * rl)) / static_cast<amrex::Real>(rl);
```

Comment:
At this point, I wonder if it would be slightly better to remove the `if` conditions on `sj`, `sk` and `sl` from these nested loops, and have them only once instead, before the loops. `sj`, `sk` and `sl` do not vary depending on `jj`, `kk` and `ll`, so in principle there would be no need to check them within these nested loops.

I think we could do something like the following:
```cpp
wj = (rj - amrex::Math::abs(j + hj - (jc + jj + hj) * rj)) / static_cast<amrex::Real>(rj);
wk = (rk - amrex::Math::abs(k + hk - (kc + kk + hk) * rk)) / static_cast<amrex::Real>(rk);
wl = (rl - amrex::Math::abs(l + hl - (lc + ll + hl) * rl)) / static_cast<amrex::Real>(rl);
```
where `hj`, `hk` and `hl` are defined, before the nested loops, as
```cpp
amrex::Real hj = (sj == 0) ? 0.5_rt : 0._rt;
amrex::Real hk = (sk == 0) ? 0.5_rt : 0._rt;
amrex::Real hl = (sl == 0) ? 0.5_rt : 0._rt;
```
What do you think?

Reply:
yes, I like it this way. Thank you! I will make the change and push it. Also, thank you for resetting the CI benchmarks!